lib/configParser.js

```diff
@@ -58,9 +90,8 @@ var ConfigParser = function() {
  */
 var merge_ = function(into, from) {
   for (var key in from) {
-    if (into[key] instanceof Object &&
-        !(into[key] instanceof Array) &&
-        !(into[key] instanceof Function)) {
+    if ((isArrayLike_(into[key]) && isArrayLike_(from[key])) || 
+        (isDictionary_(into[key]) && isDictionary_(from[key]))) {
```

Comment:
Actually, I realize that my situation above would work because optimist will parse the command line as a string. But what about:
config:
`specs: ['1.js', '2.js', '3.js']`
command line:
`--specs 1.js,2.js`